apps/cowswap-frontend/src/modules/tradeFlow/services/safeBundleFlow/safeBundleApprovalFlow.ts

```diff
@@ -59,8 +67,35 @@ export async function safeBundleApprovalFlow(
     orderParams.appData = await removePermitHookFromAppData(orderParams.appData, typedHooks)
 
     logTradeFlow(LOG_PREFIX, 'STEP 3: post order')
-    const { id: orderId, order } = await signAndPostOrder(orderParams).finally(() => {
-      callbacks.closeModals()
+    const {
+      orderId,
+      signingScheme,
+      signature,
+      orderToSign: unsignedOrder,
+    } = await wrapErrorInOperatorError(() =>
+      tradeQuote
+        .quote!.postSwapOrderFromQuote({
```

Comment:
With the check above on line 38, is the non-null assertion still needed?

Reply:
Thanks!
Fixed that: https://github.com/cowprotocol/cowswap/pull/5549/commits/e901c98458819c074859ebef7bdfbaffe4ce27e8